### ISSUE TITLE
X.509 certificate chains using md5RSA signatures not rejected

### ISSUE THREAD

**Issue by MichalStaruch** [curl/curl#1278]
### I did this
``$ curl -I "https://[test_host]"``
where X.509 chain provided by test_host was using subject certificate with md5RSA signature.

### I expected the following
Connection rejected with verification error. Support for signatures using MD5 disabled by default.

OpenSSL 1.1.0e gives following error on the same host: ``Verification error: CA signature digest algorithm too weak``.

Chrome 56 and Firefox 51 reject the connection as well, giving ``NET::ERR_CERT_WEAK_SIGNATURE_ALGORITHM`` and ``SEC_ERROR_CERT_SIGNATURE_ALGORITHM_DISABLED`` errors, respectively.

### curl/libcurl version
~~~
curl 7.52.1 (x86_64-pc-linux-gnu) libcurl/7.52.1 OpenSSL/1.0.2k zlib/1.2.8 libidn2/0.16 libpsl/0.17.0 (+libidn2/0.16) libssh2/1.7.0 nghttp2/1.18.1 librtmp/2.3
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL
~~~

### operating system
Debian Stretch RC2.

**Comment by bagder** [comment]
Yeah, there's no attempt by curl itself to catch these things. You up to providing a patch/pull request?

**Comment by MichalStaruch** [comment]
@bagder Guess I could. I will give you real answer after I look at curl sources.

**Comment by MichalStaruch** [comment]
@badger: I looked at how curl handles expired certificates (they are properly rejected) to have some starting point. It seemed that curl fowards OpenSSL errors in ossl_connect_step2(), under ``lerr != X509_V_OK`` condition.

I've looked at how OpenSSL handles it, and it turned out ``X509_V_ERR_CA_MD_TOO_WEAK`` error was added in OpenSSL 1.1.0. So I simply compiled curl against OpenSSL 1.1.0 and did the test again - connection was properly rejected, with ``curl: (60) SSL certificate problem: CA signature digest algorithm too weak`` error from OpenSSL.

So... curl itself seems to be fine (no compatibility problems with OpenSSL 1.1.0 at first look), and problem in Debian should be solved when package maintainers move to newer OpenSSL (I will contact them soon).
